pkg/v3/types/basetypes.go

```diff
@@ -0,0 +1,145 @@
+package types
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+)
+
+type UpkeepType uint8
+
+const (
+	// TODO: add type for unknown
```

Comment:
created ticket auto 4335 for this. I think this will be a tech debt item and not something we fix before launch